.github/workflows/remote-trigger.yml

```diff
@@ -42,11 +42,49 @@ jobs:
           path: e2e-system-tests
 
       - name: Set target
+        shell: bash
         run: |
-          branch_arg=${{ inputs.ref }}
-          echo "CYPRESS_BRB=https://brb-${branch_arg#refs/heads/}.cd.dbildungscloud.dev" >> $GITHUB_ENV
-          echo "CYPRESS_NBC=https://nbc-${branch_arg#refs/heads/}.cd.dbildungscloud.dev" >> $GITHUB_ENV
-          echo "CYPRESS_DEFAULT=https://default-${branch_arg#refs/heads/}.cd.dbildungscloud.dev" >> $GITHUB_ENV
+          branch_name=${{ inputs.ref }}
+          echo "input branch name: $branch_name"
+
+          # This cleaning of the branch name needs to be done the same way as in deployment. 
+          # Currently done in https://github.com/hpi-schul-cloud/dof_app_deploy/blob/main/.github/workflows/clean_workflow.yml
+          cleaned_branch_name=$(echo $branch_name | sed 's@.*/@@' | tr [A-Z] [a-z] | tr _ - | tr \. -)
+          echo "cleaned branch name: $cleaned_branch_name"
+
+          echo "CYPRESS_BRB=https://brb-${cleaned_branch_name}.cd.dbildungscloud.dev" >> $GITHUB_ENV
+          echo "Set CYPRESS_BRB=https://brb-${cleaned_branch_name}.cd.dbildungscloud.dev"
+
+          echo "CYPRESS_NBC=https://nbc-${cleaned_branch_name}.cd.dbildungscloud.dev" >> $GITHUB_ENV
+          echo "Set CYPRESS_NBC=https://nbc-${cleaned_branch_name}.cd.dbildungscloud.dev"
+
+          echo "CYPRESS_DEFAULT=https://default-${cleaned_branch_name}.cd.dbildungscloud.dev" >> $GITHUB_ENV
+          echo "Set CYPRESS_DEFAULT=https://default-${cleaned_branch_name}.cd.dbildungscloud.dev" 
+
+      - name: Check if all apps are running
+        shell: bash
+        run: |
+          check_availability () {
+            delay=10
+            max_tries=30
+            tries=0
+
+            until [ $tries -ge $max_tries ] || curl -sf $1 | jq '."services-unavailable"' | grep false > /dev/null; do
+              echo "Apps are not running. Try again in $delay seconds."
+              tries=$((tries+1))
+              sleep $delay
+            done
+    
+            if [ $tries -ge $max_tries ]; then
+              echo "Terminated checking $1 after $tries tries without success."
```

Comment:
should we not terminate here?
exit 1

Reply:
Yes, you're totally right. I forgot that. Added it now.